index.js

```diff
@@ -1,6 +1,8 @@
 var urlMapper = require('url-mapper');
-var addressbar = require('addressbar');
 var pathToRegexp = require('path-to-regexp');
+try {
+  var addressbar = require('addressbar');
+} catch(e) {}
```

Comment:
Ah, you thinking about preparing for server... you know, I am actually not sure that the router has any purpose on the server. Lets say you want to open a list of todos, you go to: `/todos`. What you want to happen is this:

```js
app.get('/todos', (req, res) => {
  db.getTodos()
    .then((todos) => {
      const state = Object.assign({}, initialState, {todos: todos});
      const controller = CerebralDummyController(state);
      const appHtml = renderToString(<Container controller={controller}><App/></Container>);
      res.render('index', {app: appHtml, state: state});
    })
});
```

So you do not care about the router in the client, you just care about the request to the server. That request should prepare some data that should be merged into the initial state of the application. Then you create a dummy controller to expose that state to the app when rendered. Then you have to inject the app and actually also the state into the html file. That way the client will render the app correctly, but also build up the initial client side state store correctly with the state from the server.

I think firing off signals etc. on the server will just be confusing. Just think about the components and inject the state they need.